pulp/apis/scip_api.py

```diff
@@ -90,17 +97,21 @@ def actualSolve(self, lp):
         if self.timeLimit is not None:
             proc.extend(['-c', 'set limits time {}'.format(self.timeLimit)])
 
-        maxNodes = self.optionsDict.get('maxNodes')
-        if maxNodes is not None:
-            proc.extend(['-c', 'set limits nodes {}'.format(maxNodes)])
+        options = self.options + self.getOptions()
+        options = list(itertools.chain(*[("-c",o) for o in options]))
+        proc.extend(options)
+
         proc.extend(self.options)
         if not self.msg:
             proc.append('-q')
         proc.extend(['-c', 'optimize', '-c', 'write solution "%s"' % tmpSol, '-c', 'quit'])
 
-        self.solution_time = -clock()
-        subprocess.check_call(proc, stdout=sys.stdout, stderr=sys.stderr)
-        self.solution_time += clock()
+        stdout = self.firstWithFilenoSupport(sys.stdout, sys.__stdout__)
+        stderr = self.firstWithFilenoSupport(sys.stderr, sys.__stderr__)
+
+        self.solution_time = -resource.getrusage(resource.RUSAGE_CHILDREN).ru_utime
+        subprocess.check_call(proc, stdout=stdout, stderr=stderr)
+        self.solution_time += resource.getrusage(resource.RUSAGE_CHILDREN).ru_utime
```

Comment:
I'm wondering why do you change the time measure? I'd ideally have the same way to measure time across solvers. How is it different from pulp's clock?

Reply:
`clock()` measures only the time for the current process, excluding the child process  that is created with `check_call`.

For `PULP_CBC_CMD` it is also not working correctly, as I tested now: 
![coin_time](https://user-images.githubusercontent.com/229571/114175561-5b122200-993a-11eb-96fc-d434cb02a37c.png)
and CBC logs
![logs](https://user-images.githubusercontent.com/229571/114175643-754c0000-993a-11eb-8ea1-c1121075aab8.png)

Same for `COIN_CMD`

Reply:
Locally changed lines  https://github.com/coin-or/pulp/blob/a587e8e69d6629b3a55b2559a08179f019085253/pulp/pulp.py#L1739-L1741
and used `resource.getrusage(resource.RUSAGE_CHILDREN).ru_utime` instead of `clock()`.
For `GLPK_CMD`,  `PULP_CBC_CMD`, `COIN_CMD`, `SCIP_CMD' I get a correct `prob.solutionTime` and without this change, it is a wrong number close to 0.

Reply:
Currently:  `clock()`
![clock](https://user-images.githubusercontent.com/229571/114185548-260bcc80-9946-11eb-8517-e27e3f6dddd7.png)
With ressource usage of child processes:
![rusage](https://user-images.githubusercontent.com/229571/114185617-3754d900-9946-11eb-8592-de9277827c59.png)

Interestingly, the `COIN_CMD` shows that the CPU time is measured, the logs of CBC show:
```
Result - Stopped on time limit

Objective value:                11.00000000
Lower bound:                    10.000
Gap:                            0.10
Enumerated nodes:               4414
Total iterations:               72247
Time (CPU seconds):             9.61
Time (Wallclock seconds):       6.50

Option for printingOptions changed from normal to all
Total time (CPU seconds):       9.66   (Wallclock seconds):       6.51
```

Reply:
I could create another PR changing this measuerement in `pulp.py` so that you could test it with other solver that you have access to.
I'm not entirely sure that every solver runs as a child process.